File: 1.cs
namespace _1;

public class Solution
{
    public int[] TwoSum(int[] nums, int target)
    {
        Dictionary<int, int> dict = new();
        for (int i = 0; i < nums.Length; i++)
        {
            int diferencia = target - nums[i];
            if (dict.ContainsKey(diferencia))
                return new[] { dict[diferencia], i };

            if (!dict.ContainsKey(nums[i]))
                dict.Add(nums[i], i);
        }

        return new[] { -99, -99 };
    }
}

File: PHP/NeetCode roadmap/Arrays & Hashing/1.php
<?php

class Solution
{
    public function twoSum(array $nums, int $target): array
    {
        $diffs = [];

        for ($i = 0; $i < count($nums); $i++) {
            $diff = $target - $nums[$i];

            if (array_key_exists($diff, $diffs)) {
                return [$diffs[$diff], $i];
            }

            $diffs[$nums[$i]] = $i;
        }

        return [];
    }
}
